### PR TITLE
executeOnceBeforeRender including optional timeout

### PR DESCRIPTION
**Pull request by RaananW**

### CODE REVIEW

File: src/babylon.scene.ts [hunk 1]
```diff
@@ -1924,6 +1924,31 @@
...
+
+        /**
+         * The provided function will run before render once and will be disposed afterwards.
+         * A timeout delay can be provided so that the function will be executed in N ms.
+         * The timeout is using the browser's native setTimeout so time percision cannot be guaranteed.
+         * @param func The function to be executed.
+         * @param timeout optional delay in ms
+         */
+        public executeOnceBeforeRender(func: () => void, timeout?: number): void {
+            if (typeof timeout === 'number') {
```
Comment (deltakosh) [inline src/babylon.scene.ts]: why not just testing timeout !== undefined?

File: src/babylon.scene.ts [hunk 2]
```diff
@@ -1924,6 +1924,31 @@
             this.onAfterRenderObservable.removeCallback(func);
         }
 
+        private _executeOnceBeforeRender(func: () => void): void {
+            let execFunc = () => {
+                func();
+                this.unregisterBeforeRender(execFunc);
```
Comment (deltakosh) [inline src/babylon.scene.ts]: I recommend using a setTimeout(0) for the unregister as you are updating the list of callbacks inside a loop thought the same callback list

File: src/babylon.scene.ts (hunk 1 above)
Comment (RaananW) [reply src/babylon.scene.ts]: I just wanted to make sure that 0 can still be provided (wanted to avoid !timeout). I can change that

Comment (deltakosh) [reply src/babylon.scene.ts]: Yep the cost of typeof compared to simple eauqlity is higher

**Pull request merged**